src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs

```diff
@@ -160,6 +182,88 @@ private void SetPanelSizes(float panel1Fraction)
         _panel2Size = string.Create(CultureInfo.InvariantCulture, $"{(1 - panel1Fraction):F3}fr");
     }
 
+    public async Task OnPageKeyDownAsync(KeyboardEventArgs args)
+    {
+        if (_splitterRef is null || !args.ShiftKey)
+        {
+            return;
+        }
+
+        if (args.Key.Equals("t", StringComparison.InvariantCultureIgnoreCase))
+        {
+            await HandleToggleOrientation();
+            return;
+        }
+
+        if (args.Key.Equals("o", StringComparison.InvariantCultureIgnoreCase) && SelectedValue is not null)
+        {
+            _internalShowDetails = !_internalShowDetails;
+            await InvokeAsync(StateHasChanged);
```

Comment:
I think we need to call the Dismiss callback here. I'm not 100% sure on whether we should do that _in addition to_ what's here or _instead of_ what's here. It seems like the latter should be sufficient as it is how the button works now - we don't actually do anything to close the details when they click the button. We just raise the dismiss callback and let the container page decide what to do (theoretically they could have a confirmation dialog or something first, though we don't do that now).

I think the lack of the Dismiss callback is why Shift+o will open the details page back up - the containing page isn't clearing the selected resource/span/etc when the user hits Shift+o. 

Also, at that point, its only ever a close, not a toggle. In which case I wonder if that shouldn't be a different shortcut. Escape? @leslierichardson95 ?

Reply:
Absolutely open to whatever shortcut you guys think is preferred. Changed the behavior to what you mention Tim, as a close shortcut.